src/infra/ucan.js

```diff
@@ -5,80 +5,114 @@ const ucans = require('ucans');
 
 const serviceDID = config.SERVICE_DID;
 
-let verify = (req, res, next) => {
-  req.requestId = uuidv4();
-  console.log('req.requestId: ', req.requestId);
-  let token = req.headers['authorization']; // Express headers are auto converted to lowercase
-  if (token && token.startsWith('Bearer ')) {
-    // Remove Bearer from string
-    token = token.slice(7, token.length);
+
+async function namespaceValidation(namespace, invokerAddress, token) {
+  try {
+    const result = await ucans.verify(token, {
+      audience: serviceDID,
+      requiredCapabilities: [
+        {
+          capability: {
+            with: { scheme: "storage", hierPart: invokerAddress },
+            can: { namespace: namespace, segments: ["CREATE", "GET"] }
+          },
+          rootIssuer: invokerAddress,
+        }
+      ],
+    });
+    return result.ok;
+  } catch (error) {
+    console.error('Error verifying UCAN with namespace:', error);
+    return false;
+  }
+}
+
+async function contractAddressValidation(contractAddress, invokerAddress, token, chainId) {
+  let invokerDid = null;
+
+  try {
```

Comment:
why use multiple try catch here .. this can be done in a single try catch right ?

Reply:
added different try catch to handle different errors with more readable logs.